rustls-mbedcrypto-provider/src/self_tests.rs

```diff
@@ -0,0 +1,175 @@
+//! This module defines self-tests. Running tests in this module (i.e., calling [`self_tests()`](self::self_tests()))
+//! at runtime can help with [FIPS 140-3] compliance.
+//!
+//! [FIPS 140-3]: (https://csrc.nist.gov/csrc/media/Projects/cryptographic-module-validation-program/documents/fips%20140-3/FIPS%20140-3%20IG.pdf)
+
+use rustls::crypto::{
+    tls12::{Prf, PrfUsingHmac},
+    tls13::{expand, Hkdf, HkdfUsingHmac},
+};
+use std::vec::Vec;
+
+// test copied from rustls repo
+/// TLS 1.2 SHA256 PRF test
+#[cfg(feature = "tls12")]
+pub fn tls12_sha256_prf_test_1() {
+    let secret = b"\x9b\xbe\x43\x6b\xa9\x40\xf0\x17\xb1\x76\x52\x84\x9a\x71\xdb\x35";
+    let seed = b"\xa0\xba\x9f\x93\x6c\xda\x31\x18\x27\xa6\xf7\x96\xff\xd5\x19\x8c";
+    let label = b"test label";
+    let expect = include_bytes!("../testdata/prf-result.1.bin");
+    let mut output = [0u8; 100];
+
+    let prf = PrfUsingHmac(&super::hmac::HMAC_SHA256);
+    prf.for_secret(&mut output, secret, label, seed);
+
+    assert_eq!(expect.len(), output.len());
+    assert_eq!(&expect[..], &output[..]);
+}
+
+/// TLS 1.2 SHA256 PRF test with `"extended master secret"`
+#[cfg(feature = "tls12")]
+pub fn tls12_sha256_prf_test_2() {
```

Comment:
I guess this test is fresh created with:
Printing the result of random input, then update the test with hardcoded expected result?

Reply:
Yes 😅